e2e/playwright/testing-samples-loading.spec.ts

```diff
@@ -200,4 +185,96 @@ test.describe('Testing in-app sample loading', () => {
       })
     }
   )
+
+  const externalModelCases = [
+    {
+      modelName: 'cylinder.kcl',
+      deconflictedModelName: 'cylinder-1.kcl',
+      modelPath: executorInputPath('cylinder.kcl'),
+    },
+    {
+      modelName: 'cube.step',
+      deconflictedModelName: 'cube-1.step',
+      modelPath: testsInputPath('cube.step'),
+    },
+  ]
+  externalModelCases.map(({ modelName, deconflictedModelName, modelPath }) => {
+    test(
+      `Load external models from local drive - ${modelName}`,
+      { tag: ['@electron'] },
+      async ({ page, homePage, scene, toolbar, cmdBar, tronApp }) => {
+        if (!tronApp) {
+          fail()
+        }
+
+        await page.setBodyDimensions({ width: 1000, height: 500 })
+        await homePage.goToModelingScene()
+        await scene.settled(cmdBar)
+        const modelFileContent = await fsp.readFile(modelPath, 'utf-8')
+        const { editorTextMatches } = await getUtils(page, test)
+
+        async function loadExternalFileThroughCommandBar(tronApp: ElectronZoo) {
+          await toolbar.loadButton.click()
+          await cmdBar.expectState({
+            commandName: 'Load external model',
+            currentArgKey: 'source',
+            currentArgValue: '',
+            headerArguments: {
+              Method: 'newFile',
+              Sample: '',
+              Source: '',
+            },
+            highlightedHeaderArg: 'source',
+            stage: 'arguments',
+          })
+          await cmdBar.selectOption({ name: 'Local Drive' }).click()
+
+          // Mock the file picker selection
+          const handleFile = tronApp.electron.evaluate(
+            async ({ dialog }, filePaths) => {
+              dialog.showOpenDialog = () =>
+                Promise.resolve({ canceled: false, filePaths })
+            },
+            [modelPath]
+          )
+          await page.getByTestId('cmd-bar-arg-file-button').click()
+          await handleFile
+
+          await cmdBar.progressCmdBar()
+          await cmdBar.expectState({
+            commandName: 'Load external model',
+            headerArguments: {
+              Source: 'local',
+              Path: modelName,
+            },
+            stage: 'review',
+          })
+          await cmdBar.progressCmdBar()
+        }
+
+        await test.step('Load the external model from local drive', async () => {
+          await loadExternalFileThroughCommandBar(tronApp)
+          // TODO: I think the files pane should auto open?
```

Comment:
This is interesting. Once I get just a bit of bandwidth I think we should talk about a future UI where file management, feature tree, and code pane aren't so disparate. Then we won't need to manage the state of the panes so much to make the UX smooth.